openedx/license-manager
# Add license-subisdy api endpoint for learners

## Description
This endpoint allows a learner to lookup the subsidy data for a given
course. The endpoint looks at the given enterprise customer, gets that
customer's active subscription plan, and checks to see if a user has an
activated license for that subscription. If so, the endpoint checks to
verify that the course is in the given subscription's enterprise
catalog. If the course is in the catalog, the user has access to that
content, so the endpoint returns metadata relevant to the subsidy the
license provides.

Link to the associated ticket: https://openedx.atlassian.net/browse/ENT-2752

## Testing considerations
- Make sure your devstack, license manager, and enterprise catalog services are all up and running
- Add an **active** subscription plan with the enterprise customer uuid & catalog uuid from enterprise-catalog admin
- Hit http://localhost:18160/api/v1/enterprise-catalogs/{enterprise_catalog_uuid/get_content_metadata.json and find a course key in the catalog's metadata to use for testing
- Hit http://localhost:18170/api/v1/license-subsidy/?enterprise_customer_uuid={enterprise_customer_uuid}&course_key={course_key} as a learner not tied to the enterprise customer
  - Verify you receive `"detail": "Missing: subscriptions.has_learner_access"` as the response
- Hit the license subsidy endpoint without either of the above query parameters and verify the view gives a 4xx response to both requests
- Hit the license subsidy endpoint with the query parameters again, but with a learner associated with the enterprise. Verify that a 200 response is given with the expected data
- Hit the license subsidy endpoint again, but this time use a `course_key` that is not in the catalog's metadata. Verify that a 404 response is given
